parse/builder/builder.go

```diff
@@ -146,7 +147,11 @@ func (b *Builder) checkDefaultAndRequired(schema *types.Schema, input map[string
 		if op.IsList() && fieldMatchesOp(field, List) && definition.IsReferenceType(field.Type) && !hasKey {
 			result[fieldName] = nil
 		} else if op.IsList() && fieldMatchesOp(field, List) && !hasKey && field.Default != nil {
-			result[fieldName] = field.Default
+			blank, err := ConvertSimpleNil(field.Type)
```

Comment:
I believe the requirement was that we only give add the key with its zero value if the field is no nullable. right?